### PR TITLE
Swift CRT Bindings

### PR DESCRIPTION
**Pull request by kneekey23**
*Description of changes:* Merge to Master from integration branch where majority of bindings were written.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by justinboswell — APPROVED** [review]
Ignoring the linux CI failures for now

**Comment by kneekey23** [comment]
ignoring failing linux tests per conversation with @JonathanHenson and @justinboswell and merging

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+awsCIoPlatformExcludes.append("source/linux")
+awsCIoPlatformExcludes.append("source/s2n")
+#elseif(Windows)
+awsCIoPlatformExcludes.append("source/posix")
+awsCIoPlatformExcludes.append("source/linux")
+awsCIoPlatformExcludes.append("source/s2n")
+awsCIoPlatformExcludes.append("source/darwin")
+#else
+awsCIoPlatformExcludes.append("source/windows")
+awsCIoPlatformExcludes.append("source/s2n")
```
Comment (JonathanHenson) [inline Package.swift:57]: this should be:

awsCIoPlatformExcludes.append("source/bsd")

My fault I know

File: Source/AwsCommonRuntimeKit/AwsCommonRuntimeKit.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  public static let version = "@AWS_CRT_SWIFT_VERSION@"
+
+  public static func initialize(allocator: Allocator = defaultAllocator) {
+    aws_http_library_init(allocator.rawValue)
+    aws_mqtt_library_init(allocator.rawValue)
+    aws_auth_library_init(allocator.rawValue)
+  }
+
+  public static func cleanUp() {
+    aws_mqtt_library_clean_up()
```
Comment (JonathanHenson) [inline Source/AwsCommonRuntimeKit/AwsCommonRuntimeKit.swift:15]: auth, mqtt, then http

File: Source/AwsCommonRuntimeKit/Utilities.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+import AwsCIo
+import struct Foundation.Date
+import struct Foundation.Data
+import class Foundation.FileHandle
+import AwsCCommon
+
+@inlinable
+func zeroStruct<T>(_ ptr: UnsafeMutablePointer<T>) {
```
Comment (JonathanHenson) [inline Source/AwsCommonRuntimeKit/Utilities.swift:10]: this can likely be used instead of the initialize stuff you had

**Review by JonathanHenson — APPROVED** [review]

File: Package.swift (hunk 1 above)
Comment (kneekey23) [reply Package.swift:57]: in addtion to whats there or instead of "source/s2n"?

File: Source/AwsCommonRuntimeKit/Utilities.swift (hunk 3 above)
Comment (kneekey23) [reply Source/AwsCommonRuntimeKit/Utilities.swift:10]: corrected

**Pull request merged**